### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by mariofriz**
First version as PR to this repo.

I have not changed the name + namespace of the package. Should I change it to something like `Nimiq\Rpc` ?

Today package is deployed to Packagist: https://packagist.org/packages/lunanimous/php-rpc-client

Happy review 👍

### CODE REVIEW

**Comment by sisou** [comment]
@svub can you please assign me to this PR, request a review from me, and add me as a maintainer for this repo or organization, thanks!

File: .gitignore
Comment (sisou) [inline .gitignore:10]: I think the `composer.lock` file should be committed to the code.

File: README.md
Comment (sisou) [inline README.md]:
```suggestion
echo $blockNumber; // displays the block number, for example 748883
```

File: README.md
Comment (sisou) [inline README.md]:
```suggestion
Check out the [Nimiq RPC specs](https://github.com/nimiq/core-js/wiki/JSON-RPC-API) for behind the scene RPC calls.
```

File: README.md
Comment (sisou) [inline README.md]:
```suggestion
All done, happy coding!
```

File: docs/README.md
Comment (sisou) [inline docs/README.md:63]: Is it standard that the `__construct` method is documented like this? Is it possible to document it as `new .../NimiqClient()`?

File: docs/README.md
Comment (sisou) [inline docs/README.md]: The opening and closing triangle brackets `&lt;` and `&gt;` are displayed as their HTML code in the readme, not as brackets. Is it possible to fix that? This is happening in multiple places in the /docs README...

File: docs/README.md
Comment (sisou) [inline docs/README.md:52]: Would it be possible to include the Models documentation into this README? So that I can see which fields a `Model/Transaction` or a `Model/Block` has, without opening the `src` file?

EDIT: After some research, it seems the markdown-template that you use does only support classes and methods, not vars. Can this be added?

File: src/Models/Account.php
Comment (sisou) [inline src/Models/Account.php]: For Vesting Contracts and HTLCs, the result also includes the respective extra fields for those contracts:
https://github.com/nimiq/core-js/blob/03cd33f2bffb9409246427210c0223168adadd1f/clients/nodejs/modules/JsonRpcServer.js#L650-L677

These are not represented here. They should be made available, either on this model or on custom models based on the account type.

File: src/functions.php
Comment (sisou) [inline src/functions.php]: The official JS implementation in [`Policy.js`](https://github.com/nimiq/core-js/blob/3e6cb39625a278dcd9f4406b9ed5dff55b17b36d/src/main/generic/consensus/Policy.js#L8) also rounds here, to be sure that the result contains no decimals. You should either also round, or cast the output to `(int)` to make sure.

File: src/functions.php
Comment (sisou) [inline src/functions.php]: Maybe it's valuable to make sure the input is an `int`, so that we can be sure the result has a maximum of 5 decimal places.

File: tests/fixtures/getAccount/account.json
Comment (sisou) [inline tests/fixtures/getAccount/account.json]: Please include tests for contract account types.

File: tests/fixtures/accounts/accounts.json
Comment (sisou) [inline tests/fixtures/accounts/accounts.json]: Please include tests for contract account types.

**Review by sisou — CHANGES_REQUESTED** [review]
This is **very nice** code! Especially all the tests are impressive!

My main questions revolve around two non-code things:
- Would it be possible to also document the `Models` automatically, including their `@var`s? That would help a lot without having to refer to src files.
- Please add support and tests for Vesting and HTL contracts, as they contain more properties than the basic account, which are important as well.

Well done! :+1:

File: .gitignore
Comment (mariofriz) [reply .gitignore:10]: I wasn't sure about this but it seems it's not recommended:
https://getcomposer.org/doc/02-libraries.md#lock-file

File: docs/README.md
Comment (mariofriz) [reply docs/README.md:63]: This is the way phpdocumenter describe a constructor, just like any other method but it's a "magic" method. Other projects also use it: https://laravel.com/api/5.8/Illuminate/Auth/AuthManager.html#method___construct

I don't think this is a blocker, if I have some time I'll create my own phpdocumentor plugin to output markdown and have the "new" syntax.

File: src/Models/Account.php
Comment (mariofriz) [reply src/Models/Account.php]: Just to confirm, would this be a valid HTLC response ?

```json
{
    "jsonrpc": "2.0",
    "result": {
        "id": "4974636bd6d34d52b7d4a2ee4425dc2be72a2b4e",
        "address": "NQ46 NTNU QX94 MVD0 BBT0 GXAR QUHK VGNF 39ET",
        "balance": 1000000000,
        "type": 2,
        "sender": "d62d519b3478c63bdd729cf2ccb863178060c64a",
        "senderAddress": "NQ53 SQNM 36RL F333 PPBJ KKRC RE33 2X06 1HJA",
        "recipient": "f5ad55071730d3b9f05989481eefbda7324a44f8",
        "recipientAddress": "NQ41 XNNM A1QP 639T KU2R H541 VTVV LUR4 LH7Q",
        "hashRoot": "df331b3c8f8a889703092ea05503779058b7f44e71bc57176378adde424ce922",
        "hashAlgorithm": 1,
        "hashCount": 1,
        "timeout": 1105605,
        "totalAmount": 1000000000
    },
    "id": 1
}
```

Comment (sisou) [reply src/Models/Account.php]: Yes, looks correct.

**Comment by mariofriz** [comment]
Thanks for the review, I fixed all comments.
Please do not merge yet though, I still have to:

- I'll rename the package to NimiqCommunity\RpcClient.
- Figure out how to documents models

File: src/NimiqClient.php
Comment (sisou) [inline src/NimiqClient.php]: Great! However, I belive when using `return`, you don't need to `break` anymore ;)

**Review by sisou — APPROVED** [review]

**Comment by mariofriz** [comment]
Sorry this took some time, I was quite busy.

I renamed the package to NimiqCommunity/RpcClient. 
Once the PR goes in I'll hook it up to packagist so that it's available on composer. 

Model/Constant documentation is not possible ATM, I would need to create a custom phpdoc template. Is this a blocker for the PR ?

**Review by sisou — APPROVED** [review]
I think this is great now and can be merged! :+1: :rocket:

**Pull request merged**
